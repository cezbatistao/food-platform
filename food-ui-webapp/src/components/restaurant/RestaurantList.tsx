import { useSelector } from 'react-redux';

import Grid from '@material-ui/core/Grid';

import RestaurantCard from './RestaurantCard';

import { RootState } from '../../gateway';
import Restaurant from '../../domain/Restaurant';

type RestaurantState = {
  restaurants: Restaurant[], 
  loading: boolean
}

const RestaurantList = () => {

  const { loading, restaurants }: RestaurantState = useSelector((
    state: RootState
  ) => state.restaurant);

  return (
    <Grid container spacing={3}>
      {
        restaurants.map(restaurant => {
          return(
            <Grid key={ restaurant.id } item xs={12} sm={4}>
              <RestaurantCard 
                name={ restaurant.name } 
                logo={ restaurant.logo }
                description={ restaurant.description }
                address={ restaurant.address }
              />
            </Grid>
          );
        })
      }
    </Grid>
  );
}

export default RestaurantList;