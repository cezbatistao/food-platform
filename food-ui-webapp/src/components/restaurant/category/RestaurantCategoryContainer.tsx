import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation, useHistory } from "react-router-dom";
import queryString from 'query-string';
import { Location } from 'history';

import RestaurantCategory from './RestaurantCategory';
import Category from '../../../domain/Category';
import { RootState } from '../../../gateway';

import { getCategories } from '../../../gateway/actions/category.actions';

type CategoryState = {
  categories: Category[], 
  loading: boolean
}

const RestaurantCategoryContainer = () => {

  // const location = useLocation();
  // let history = useHistory();

  // const getCategoryFromQueryString = (location: Location<unknown>) => {
  //   const parsed = queryString.parse(location.search);
  //   return parsed.category ? parsed.category as string : undefined;
  // }

  // const handleChangeCategory = (event: React.ChangeEvent<{ value: unknown }>) => {
  //   const categorySelected = event.target.value as string;
  //   history.push({
  //     search: `?category=${categorySelected}`
  //   });
  // }

  // const categoryQueryString = getCategoryFromQueryString(location);

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getCategories());
    // dispatchGetRestaurantsByCategory(categoryQueryString as string);
    
  }, [dispatch]);

  const { loading, categories }: CategoryState = useSelector((
    state: RootState
  ) => state.category);

  // const dispatchGetRestaurantsByCategory = (category: string) => {
  //   if(category) {
  //     dispatch(getRestaurantsByCategory(category));
  //   }
  // }

  return (
    <RestaurantCategory 
      loading={loading} 
      categories={categories} 
      dispatch={dispatch} 
    />
  );
}

export default RestaurantCategoryContainer;