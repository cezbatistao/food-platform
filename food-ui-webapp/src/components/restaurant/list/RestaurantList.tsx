import { Fragment } from 'react';
import { useSelector } from 'react-redux';

import RestaurantCard from '../card/RestaurantCard';
import { RootState } from '../../../gateway';
import Restaurant from '../../../domain/Restaurant';

type RestaurantState = {
  restaurants: Restaurant[], 
  loading: boolean
}

const RestaurantList = () => {

  const { loading, restaurants }: RestaurantState = useSelector((
    state: RootState
  ) => state.restaurant);
  
  return (
    <div className="row row-cols-1 row-cols-md-5 g-4">
      {
        restaurants.map(restaurant => {
          return(
            <RestaurantCard 
              uuid={ restaurant.uuid } 
              name={ restaurant.name } 
              logo={ restaurant.logo }
              description={ restaurant.description }
              address={ restaurant.address }
            />
          );
        })
      }
    </div>
  );
}

export default RestaurantList;