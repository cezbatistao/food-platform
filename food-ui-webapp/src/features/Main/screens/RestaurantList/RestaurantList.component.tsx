import { useState, useEffect, useContext } from "react";

import CategoryContext from '../../contexts/RestaurantCategory/RestaurantCategory.context';

import { fetchRestaurantsByCategory } from "../../../../services/Restaurant.services";
import RestaurantCard from "../../components/RestaurantCard/RestaurantCard.component";
import Restaurant from "../../../../services/Restaurant.model";

const RestaurantList = () => {

  const { category } = useContext(CategoryContext);

  const [restaurants, setRestaurants] = useState<Restaurant[]>([]);
  const [loading, setLoading] = useState<boolean>(true);

  useEffect(() => {
    const getRestaurantsFromCategory = async () => {
      if(category) {
        const restaurantsFromCategory: Restaurant[] = await fetchRestaurantsByCategory(category);
        setRestaurants(restaurantsFromCategory);
        setLoading(false);
      }
    }

    setLoading(true);
    getRestaurantsFromCategory();
  }, [category]);

  return (
    <div key="restaurant-list" className="row row-cols-1 row-cols-md-5 g-4">
      {!loading &&
        restaurants.map(restaurant => {
          return(
            <RestaurantCard 
              key={ restaurant.uuid } 
              uuid={ restaurant.uuid } 
              name={ restaurant.name } 
              logo={ restaurant.logo }
              description={ restaurant.description }
              address={ restaurant.address }
            />
          );
        })
      }
    </div>
  );
}
  
export default RestaurantList;