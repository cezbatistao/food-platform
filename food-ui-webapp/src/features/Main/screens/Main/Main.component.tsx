import RestaurantCategory from "../RestaurantCategory/RestaurantCategory.component";
import RestaurantList from '../RestaurantList/RestaurantList.component';
import CategoryContext from '../../contexts/RestaurantCategory/RestaurantCategory.context';

import { useState } from "react";

const Main = () => {

  const [category, setCategory] = useState('');

  return (
    <CategoryContext.Provider value={{ category, setCategory }}>
      <div className="container">
        <div className="row">
          <div className="col-6 col-md-4">
            <RestaurantCategory />
          </div>
        </div>
      </div>
      <br />
      <div className="container">
        <div className="row">
          <div className="col-8 col-md-12">
            <RestaurantList />
          </div>
        </div>
      </div>
    </CategoryContext.Provider>
  );
}

export default Main;