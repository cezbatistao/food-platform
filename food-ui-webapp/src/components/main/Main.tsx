import RestaurantList from '../restaurant/list/RestaurantList';
import RestaurantCategoryContainer from '../restaurant/category/RestaurantCategoryContainer';

const Main = () => {

  return (
    <>
      <div className="container">
        <div className="row">
          <div className="col-6 col-md-4">
            <RestaurantCategoryContainer />
          </div>
        </div>
      </div>
      <br />
      <div className="container">
        <div className="row">
          <div className="col-8 col-md-12">
            <RestaurantList />
          </div>
        </div>
      </div>
    </>
  );
}

export default Main;