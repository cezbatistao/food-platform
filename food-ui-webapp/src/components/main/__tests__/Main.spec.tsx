import { shallow } from "enzyme";

import Main from "../Main";

import RestaurantCategoryContainer from "../../restaurant/category/RestaurantCategoryContainer";
import RestaurantList from "../../restaurant/list/RestaurantList";

describe('Main', () => {

  it("renders the main", () => {
    const result = shallow(<Main />);
    expect(result).toMatchSnapshot();
  });

  it("renders the components RestaurantCategory and RestaurantList", () => {
    const result = shallow(<Main />);
    expect(result.find(RestaurantCategoryContainer).length).toBe(1);
    expect(result.find(RestaurantList).length).toBe(1);
  });

});