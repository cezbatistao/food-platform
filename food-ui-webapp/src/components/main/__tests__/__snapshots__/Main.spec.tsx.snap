// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders the main 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-6 col-md-4"
      >
        <RestaurantCategoryContainer />
      </div>
    </div>
  </div>
  <br />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-8 col-md-12"
      >
        <RestaurantList />
      </div>
    </div>
  </div>
</Fragment>
`;
